### PR TITLE
[GDB-9922] Fixed memory utilization alarm to monitor the Auto scaling group

### PR DESCRIPTION
**Pull request by simeonzhekofff**
## Description

Fixed memory utilization alarm to be able to monitor the auto-scaling group.
Added option to enable detailed monitoring to be able to parse data every minute.

## Related Issues

[GDB-9922]

## Changes

Fixed memory utilization alarm to be able to monitor the auto-scaling group.
Added option to enable detailed monitoring to be able to parse data every minute.

## Checklist

- [X] I have tested these changes thoroughly.
- [X] My code follows the project's coding style.
- [X] I have added appropriate comments to my code, especially in complex areas.
- [X] All new and existing tests passed locally.

[GDB-9922]: https://ontotext.atlassian.net/browse/GDB-9922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/monitoring/alarms.tf [hunk 1]
```diff
@@ -89,9 +87,10 @@ resource "aws_cloudwatch_metric_alarm" "graphdb_memory_utilization" {
   namespace   = "CWAgent"
   unit        = "Percent"
 
-  //dimensions = {
-  // InstanceId = data.aws_instances.filtered_instances.ids[count.index]
-  // }
+  dimensions = {
+    AutoScalingGroupName = var.resource_name_prefix
+
```
Comment (mihailradkov) [inline modules/monitoring/alarms.tf:92]: Има!

File: modules/graphdb/main.tf [hunk 2]
```diff
@@ -53,6 +53,9 @@ resource "aws_launch_template" "graphdb" {
   instance_type = var.ec2_instance_type
   key_name      = var.ec2_key_name != null ? var.ec2_key_name : null
   user_data     = var.ec2_userdata_script
+  monitoring {
```
Comment (viktor-ribchev) [inline modules/graphdb/main.tf:56]: Няма!

File: modules/graphdb/main.tf [hunk 3]
```diff
@@ -53,6 +53,9 @@ resource "aws_launch_template" "graphdb" {
...
+    enabled = var.enable_detailed_monitoring
+  }
   vpc_security_group_ids = [
```
Comment (viktor-ribchev) [inline modules/graphdb/main.tf:59]: Няма!

File: variables.tf [hunk 4]
```diff
@@ -278,6 +278,12 @@ variable "monitoring_sns_protocol" {
   default     = "email"
 }
 
+variable "monitoring_enable_detailed_monitoring" {
+  description = "Enable or disable detailed monitoring"
```
Comment (mihailradkov) [inline variables.tf:282]: Could probably add a bit more explanation for this variable here or as an example in the readme.

**Review by viktor-ribchev — APPROVED** [review]

File: variables.tf [hunk 5]
```diff
@@ -278,6 +278,12 @@ variable "monitoring_sns_protocol" {
   default     = "email"
 }
 
+variable "monitoring_enable_detailed_monitoring" {
+  description = "If true, the launched EC2 instance will have detailed monitoring enabled"
```
Comment (mihailradkov) [inline variables.tf:282]: If this is only for ec2, maybe rename the variable to `monitoring_enable_detailed_instance_monitoring`

**Review by mihailradkov — APPROVED** [review]

**Pull request merged**